### PR TITLE
feat(connector): [Stax] Use connector_response_reference_id as reference to merchant

### LINKED ISSUE #2348: [FEATURE]: [Stax] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by ManobhavSachan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This pull request fixes #2348 issue in STAX and uses connector_response_reference_id as reference to merchant.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Fixes https://github.com/juspay/hyperswitch/issues/2348

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/stax/transformers.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ pub struct StaxPaymentsRequest {
     is_refundable: bool,
     pre_auth: bool,
     meta: StaxPaymentsRequestMetaData,
+    reference: String,
```
Comment (srujanchikke) [inline crates/router/src/connector/stax/transformers.rs:26]: Hi @ManobhavSachan  ,

In reference to issue https://github.com/juspay/hyperswitch/issues/2348 , we need to adjust the reference field mapping based on the API documentation of the connector. Here merchant reference id would be `idempotency_id `

You can find the reference for Stax API documentation [here](https://staxapi.docs.apiary.io/#reference/0/payments/charge-a-payment-method)

File: crates/router/src/connector/stax/transformers.rs [hunk 2]
```diff
@@ -282,6 +285,7 @@ pub struct StaxPaymentsResponse {
     child_captures: Vec<StaxChildCapture>,
     #[serde(rename = "type")]
     payment_response_type: StaxPaymentResponseTypes,
+    reference: Option<String>,
```
Comment (srujanchikke) [inline crates/router/src/connector/stax/transformers.rs:288]: Same here 
```suggestion
    idempotency_id: Option<String>,
```

File: crates/router/src/connector/stax/transformers.rs (hunk 1 above)
Comment (ManobhavSachan) [reply crates/router/src/connector/stax/transformers.rs]: Thank You for the feedback @srujanchikke, I have made the necessary changes.

File: crates/router/src/connector/stax/transformers.rs (hunk 2 above)
Comment (ManobhavSachan) [reply crates/router/src/connector/stax/transformers.rs]: Thank You for the feedback @srujanchikke, I have made the necessary changes.

**Review by srujanchikke — DISMISSED** [review]

**Comment by prasunna09** [comment]
Hi @ManobhavSachan Can you please fix the failing checks?

**Review by prasunna09 — APPROVED** [review]

**Comment by ManobhavSachan** [comment]
@prasunna09, I have fixed the failing checks.

**Review by srujanchikke — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @ManobhavSachan!

**Comment by VedantKhairnar** [comment]
Hey @ManobhavSachan  :wave:,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**